sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesId.cs

```diff
@@ -0,0 +1,157 @@
+
+        internal (bool, object) Key3 => _key3;
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 key.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 key.
+        /// </remarks>
+        public TimeSeriesId(object key1)
+        {
+            _key1 = (true, key1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 keys.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <param name="key2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 3 keys.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <param name="key2">The second key that identifies a Time Series instance.</param>
+        /// <param name="key3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2, object key3)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+            _key3 = (true, key3);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 string key.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 string key.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1)
+        {
+            _key1 = (true, stringKey1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 string keys.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 string keys.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1, string timeSeriesIdStringProp2)
```

Comment:
rename first param to timeSeriesIdStringProp1?

Reply:
I believe we intended to go with the first parameter naming style, but you astutely noticed we aren't using the style for both params, so at least one needs to be fixed.

Reply:
Good catch Binal.